daemon/src/cleanup.rs

```diff
@@ -0,0 +1,43 @@
+use crate::db::{insert_new_cfd_state_by_order_id, load_all_cfds};
+use crate::model::cfd::{CfdState, CfdStateCommon};
+use anyhow::Result;
+use sqlx::SqlitePool;
+use std::time::SystemTime;
+
+pub async fn transition_non_continue_cfds_to_safely_aborted(db: SqlitePool) -> Result<()> {
+    let mut conn = db.acquire().await?;
+
+    let cfds = load_all_cfds(&mut conn).await?;
+
+    // Handle about existing cfds on startup
+    for cfd in cfds {
+        match cfd.state {
+
+            // Transition all Cfds in early states that cannot be continued to SafelyAborted
+            CfdState::OutgoingOrderRequest { .. }
+            | CfdState::IncomingOrderRequest { .. }
+            | CfdState::Accepted { .. }
+            | CfdState::ContractSetup { .. } => {
```

Comment:
Doesn't this imply that `SafelyAborted` is actually just `SetupFailed`? `SafelyAborted` sounds like it was a conscious decision to abort but really we didn't have a choice here.

Reply:
Similar to the `filter_map`, you could shorted this a lot by moving the selection of states into a predicate function on the CFD.